src/Stratis.Bitcoin.Features.Consensus.Tests/CoinViews/CoinviewTests.cs

```diff
@@ -120,9 +120,13 @@ public async Task TestRewindAsync()
 
             await this.ValidateCoinviewIntegrityAsync(outPoints);
 
-            for (int i = 0; i < addChangesTimes; i++)
+            var random = new Random();
+
+            while (currentHeight > 1)
```

Comment:
Is this going to loop until random returns a 0?

Reply:
Yes we keep rewinding between >0% and 100% of the remaining blocks until there is nothing left to rewind.

`currentHeight = random.Next(currentHeight - 1) + 1;` will always reduce `currentHeight`.